File: 0027-remove-element/0027-remove-element.c
int removeElement(int* nums, int numsSize, int val) {
    int i = 0; // 결과 배열의 인덱스
    for (int j = 0; j < numsSize; j++) {
        if (nums[j] != val) {
            nums[i] = nums[j];
            i++;
        }
    }
    return i; // 새로운 배열의 길이 반환
}

File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0  # 결과 리스트의 인덱스
        for j in range(len(nums)):
            if nums[j] != val:
                nums[i] = nums[j]
                i += 1
        return i  # 새로운 리스트의 길이 반환
